src/components/kytos/napp/Toolbar.vue

```diff
@@ -1,93 +1,118 @@
 <template>
- <div class='k-toolbar' >
-  <component v-show="active == (index+1)"
-             v-for="(component, index) in inner_components"
-             :is="component.name"
-             v-bind:key="component.name">
-  </component>
- </div>
-</template>
-
-<script>
-import Vue from 'vue'
-import httpVueLoader from "./httpVueLoader.js"
-
-export default {
-  name: 'k-toolbar',
-  props: ["active", "compacted", "components"],
-  data () {
-    return {
-     url: this.$kytos_server+ 'ui/k-toolbar',
-     template: null,
-     inner_components: this.components || [] ,
-    }
-  },
-  render: function(createElement){
-    if (this.template) return this.template();
-  },
-  created() {
-      var self = this
-      $.get({
-        url: this.url,
-        datatype: 'json',
-        async: true,
-        cache: false,
-        success: function(data) {
-          self.inner_components = self.inner_components.concat(data);
-        }
-      }).always(function(){
-          self.load_components()
-          setTimeout(self.load_icons, 2000)
-      })
-  },
-  methods: {
-    load_icons () {
-      var self = this
-      var components  = $('.k-toolbar .k-toolbar-item')
-      $.each(components, function(index, component){
-          self.inner_components[index].icon = component.getAttribute('icon')
-          self.inner_components[index].tooltip = component.getAttribute('tooltip')
-      })
-      self.$emit('update:components', self.inner_components)
-    },
-    load_components (){
-      var self = this
-      $.each(self.inner_components, function(index, component){
-        if('url' in component){
-          // random is needed to avoid cache of components.
-          var url = self.$kytos_server+component.url+"?random="+Math.random()
-          Vue.component(component.name, httpVueLoader(url))
-        }
-      })
-    }
+  <div class='k-toolbar' >
+   <component v-show="active == (index+1)"
+              v-for="(component, index) in inner_components"
+              :is='component.name'
+              v-bind:key="component.name">
+   </component>
+  </div>
+ </template>
+ 
+ <script>
+ import Vue from 'vue'
+ import httpVueLoader from "./httpVueLoader.js"
+ import { loadModule } from "vue3-sfc-loader"
+ 
+ const options = {
+ 
+   moduleCache: {
+     vue: Vue
+   },
+ 
+   getFile(url) {
+ 
+     return fetch(url).then(response => response.ok ? response.text() : Promise.reject(response));
```

Comment:
Did you have the chance to try out and explore when the promise gets rejected? 

Wondering how it'll be handled.

Reply:
Nope, ill check it out to see what happens.